frontend/src/metabase/static-viz/components/Legend/Legend.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import { Group } from "@visx/group";
+import { Text } from "metabase/static-viz/components/Text";
+import { PositionedLegendItem } from "./types";
+
+import { LEGEND_CIRCLE_MARGIN_RIGHT, LEGEND_CIRCLE_SIZE } from "./constants";
+
+type LegendProps = {
+  top: number;
+  left: number;
+  lineHeight: number;
+  fontSize: number;
+  fontWeight: number;
+  items: PositionedLegendItem[];
+};
+
+export const Legend = ({
+  top,
+  left,
+  lineHeight,
+  fontSize,
+  fontWeight,
+  items,
+}: LegendProps) => {
+  return (
+    <Group left={left} top={top}>
+      {items.map((item, index) => {
+        const { name, color, left, top } = item;
+        const radius = LEGEND_CIRCLE_SIZE / 2;
+        const textX = LEGEND_CIRCLE_SIZE + LEGEND_CIRCLE_MARGIN_RIGHT;
+
+        return (
+          <Group left={left} top={top} key={name}>
```

Comment:
It's technically possible to have the same series name, so you'll get a runtime error. Though I don't think anyone is going to do this.

![image](https://user-images.githubusercontent.com/1937582/198240495-f872567a-0cd0-4915-87ef-8a1485b7c515.png)

![image](https://user-images.githubusercontent.com/1937582/198240568-6018a9fd-8fad-43b9-b9b6-3690528cb039.png)

Reply:
Good catch!